stackslib/src/chainstate/nakamoto/coordinator/tests.rs

```diff
@@ -374,7 +386,8 @@ fn test_simple_nakamoto_coordinator_1_tenure_10_blocks() {
 /// Mine a 10 Nakamoto tenures with 10 Nakamoto blocks
 #[test]
 fn test_simple_nakamoto_coordinator_10_tenures_10_blocks() {
```

Comment:
This test is currently failing. Seems to be failing on third iteration of the loop. 

```
WARN [1700578532.327215] [stackslib/src/chainstate/burn/db/sortdb.rs:2038] [chainstate::nakamoto::coordinator::tests::test_simple_nakamoto_coordinator_10_tenures_10_blocks] reward start height 41 for reward cycle 8 does not have a sorition from 54538e85cabb2462772bb668b6a8d1e967d837c1060788fd289a295c93424c49
```

Jude has discovered that it is off by one, looking at start height 41 when it should be at height 40. Why this off by one only happens in this test on the third iteration is unknown still.

Reply:
Found part of the problem: find_prepare_phase_sortitions was incorrectly adding the sortition ID of one BEFORE the prepare phase. Still failing at replay (I think because it tries to access the aggregate public key when it is not set)